### PR TITLE
Add changes to HLD after meetings on 24.10 and 29.10

### PR DESCRIPTION
**Pull request by GustawB**
Add a new version after team consultation.

### CODE REVIEW

**Comment by SzymonKozl** [comment]
are we ready to merge this? Next updates could be in separate PRs @GustawB

**Comment by GustawB** [comment]
@SzymonKozl We can merge, but first I will have to fix tex compilation (no new content)

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -22,11 +22,11 @@ jobs:
         run: |
           git config --global user.name "github-actions[bot]"
           git config --global user.email "github-actions[bot]@users.noreply.github.com"
-          if git diff -s --exit-code docs/HLD.pdf; then
+          if git diff -s --exit-code docs/HLD.tex; then
```
Comment (SzymonKozl) [inline .github/workflows/main.yml:25]: Add also check for refs.bib
```suggestion
          if git diff -s --exit-code docs/HLD.tex docs/refs.bib; then
```

**Review by SzymonKozl — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (GustawB) [reply .github/workflows/main.yml]: @SzymonKozl done

**Review by SzymonKozl — APPROVED** [review]

**Pull request merged**